kore/src/Kore/ASTVerifier/AttributesVerifier.hs

```diff
@@ -138,11 +153,30 @@ verifyNoHookAttribute attributes = do
 
 verifyNoHookedSupersort
     :: MonadError (Error VerifyError) error
-    => [Kore.Attribute.Sort.Sort]
+    => IndexedModule Verified.Pattern Attribute.Symbol attrs
+    -> Attribute.Axiom SymbolOrAlias VariableName
+    -> [Subsort.Subsort]
     -> error ()
-verifyNoHookedSupersort supersortsAtts = do
-    let isHooked = getHasDomainValues . hasDomainValues <$> supersortsAtts
-    when (or isHooked) $ koreFail "Hooked sorts may not have subsorts."
+verifyNoHookedSupersort indexedModule axiom subsorts = do
+    let isHooked =
+            getHasDomainValues . hasDomainValues
+            . getSortAttributes indexedModule
+            . Subsort.supersort
+        hookedSubsort = find isHooked subsorts
+    for_ hookedSubsort
+        $ \s -> koreFail
+            $ unlines
+                [ "Hooked sorts may not have subsorts."
+                , "hooked sort:"
+                , show . unparse $ Subsort.supersort s
+                , "its subsort:"
+                , show . unparse $ Subsort.subsort s
+                , "Location in the Kore file:"
+                , show . prettyPrintAstLocation
+                    $ locationFromAst (Subsort.supersort s)
+                , "Location in the original K file: "
+                , show . pretty $ Attribute.sourceLocation axiom
+                ]
```

Comment:
I think `s` should have a more descriptive name, and maybe there could be less indentation here, something like this:

```suggestion
    for_ hookedSubsort $ \sort ->
        koreFail . unlines $
            [ "Hooked sorts may not have subsorts."
            , "hooked sort:"
            , show . unparse $ Subsort.supersort sort
            , "its subsort:"
            , show . unparse $ Subsort.subsort sort
            , "Location in the Kore file:"
            , show . prettyPrintAstLocation
                $ locationFromAst (Subsort.supersort sort)
            , "Location in the original K file: "
            , show . pretty $ Attribute.sourceLocation axiom
            ]
```